### PR TITLE
Fixed ordered list item matching

### LINKED ISSUE #339: Numbered list is automatically created when typing a 1 at the beginning of a line
Reproducible on https://superlist.com/SuperEditor/

Current behavior: As soon as you start a line of text with "1" followed by a blank space, it will create a numbered list.

Expectation: A numbered list should get created if I type "1." followed by a blank space, similar to how it works for example in Apple Notes

Kapture 2021-11-02 at 16 44 59

### PR DESCRIPTION
**Pull request by wilsonowilson**
Closes #339

### CODE REVIEW

File: super_editor/lib/src/default_editor/common_editor_operations.dart [hunk 1]
```diff
@@ -1276,7 +1276,7 @@ class CommonEditorOperations {
     final hasUnorderedListItemMatch = unorderedListItemMatch.hasMatch(textBeforeCaret);
 
     // We want to match "1 ", " 1 ", " 1. ".
```
Comment (creativecreatorormaybenot) [inline super_editor/lib/src/default_editor/common_editor_operations.dart:1278]: The comment needs to be updated as well.

Comment (wilsonowilson) [reply super_editor/lib/src/default_editor/common_editor_operations.dart]: Nice catch! Thanks!

**Comment by matthew-carroll** [comment]
@wilsonowilson please set me as the reviewer for all super_editor work - I'm the only one who should be approving PRs.

**Comment by matthew-carroll** [comment]
Also @wilsonowilson please include a useful description with all PRs. What was the problem? How did you fix it? What might a reviewer want to know before looking at the code.

**Comment by matthew-carroll** [comment]
WRT tests, the higher level areas of the editor have been lax with tests, but we should start to reverse that trend.

Can you try writing appropriate tests to verify this behavior? The tests should be as narrow as possible, while still effectively validating the behavior.

**Comment by matthew-carroll** [comment]
@wilsonowilson do you plan to finish this work?

**Comment by wilsonowilson** [comment]
Hey @matthew-carroll, I'll add some tests later today

**Comment by britannio** [comment]
@matthew-carroll 
How about this regex? `^\s*[1][.)]\s+$`
image

It matches "1. " and "1) " but not "1 ".

**Comment by britannio** [comment]
Or more concisely: `^\s*1[.)]\s+$`
image

**Comment by matthew-carroll** [comment]
@britannio I updated this work to use your latest regex. I can't seem to push my branch to this PR, so I'm going to close this PR and open a new one.

**Pull request closed without merging**